curriculum/challenges/english/25-front-end-development/lab-availability-table/66b36358ed4f261d64840c24.md

```diff
@@ -0,0 +1,601 @@
+---
+id: 66b36358ed4f261d64840c24
+title: Build an Availability Table
+challengeType: 14
+dashedName: build-an-availability-table
+---
+
+# --description--
+
+**Objective:** Fulfill the user stories below and get all the tests to pass to complete the lab. 
+
+**User Stories:**
+
+1. You should have a table with at least three columns and five rows, headers included.
+1. Cells in the first row should be table headers.
+1. Cells in the first column should be table headers containing time values and they should have the `class` of `time`.
+1. All of the data cells should have the `class` of `available-#`, where `#` is a number from `0` to `5` that specifies the number of available people at that time.
+1. In your `root` selector, you should define CSS variables named `--color#`, where `#` is a number from `0` to `5`, and used them to give a background color to the corresponding `.available-#` elements.
+1. In your `root` selector, you should define CSS variables named `--solid-border` and `--dashed-border`. You should use them to style the bottom borders of your data cells alternating between solid and dashed borders, depending on the row. Give the rows either the class of `sharp` or `half` to style them.
+1. You should have a `div` element with the `id` of `legend`. It should contain a `span` element with the text of `Availability` and a `div` element with the `id` of `legend-gradient`.
+1. You should give the `#legend-gradient` element a linear gradient that change from `--color0` to `--color5`. Each color value should have two color-stops (expressed in percentage) to make the transition from one color to the following color a hard line.
+
+# --hints--
+
+You should have a `table` element.
+
+```js
+assert(!!$('table').length);
+```
+
+You should have only one `table` element.
+
+```js
+assert($('table').length === 1);
+```
+
+Your table should have at least `5` rows.
+
+```js
+assert.isAtLeast(document.querySelectorAll('tr')?.length, 5);
+```
+
+Your table should have at least `3` columns.
+
+```js
+const rows = [...document.querySelectorAll('tr')];
+assert(!!rows.length);
+rows.forEach((row) => {
+    assert.isAtLeast(row.children.length, 3);
+})
+```
+
+The first row in your table should contain at least three `th` elements.
+
+```js
+assert.isAtLeast(document.querySelectorAll('tr:first-child>th').length, 3)
+```
+
+The first row in your table should not contain `td` elements.
```

Comment:
This test is always passing.

Reply:
Now it should fail until a `tr` is added to the page.